src/ScheduledTask/Webhook/AuthorisationWebhookHandler.php

```diff
@@ -0,0 +1,135 @@
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2022 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\ScheduledTask\Webhook;
+
+use Adyen\Shopware\Entity\Notification\NotificationEntity;
+use Adyen\Shopware\Service\CaptureService;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionEntity;
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionStateHandler;
+use Shopware\Core\Framework\Context;
+
+class AuthorisationWebhookHandler implements WebhookHandlerInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var CaptureService
+     */
+    private $captureService;
+
+    /**
+     * @var OrderTransactionStateHandler
+     */
+    private $orderTransactionStateHandler;
+
+    /**
+     * @param CaptureService $captureService
+     * @param OrderTransactionStateHandler $orderTransactionStateHandler
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        CaptureService $captureService,
+        OrderTransactionStateHandler $orderTransactionStateHandler,
+        LoggerInterface $logger
+    ) {
+        $this->captureService = $captureService;
+        $this->orderTransactionStateHandler = $orderTransactionStateHandler;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param OrderTransactionEntity $orderTransactionEntity
+     * @param NotificationEntity $notificationEntity
+     * @param string $state
+     * @param string $currentTransactionState
+     * @param Context $context
+     * @return OrderTransactionEntity
+     * @throws \Adyen\Shopware\Exception\CaptureException
+     */
+    public function handleWebhook(
+        OrderTransactionEntity $orderTransactionEntity,
+        NotificationEntity $notificationEntity,
+        string $state,
+        string $currentTransactionState,
+        Context $context
+    ): OrderTransactionEntity {
+        if ($notificationEntity->isSuccess() && $state !== $currentTransactionState) {
+            $this->handleSuccessfulNotification($orderTransactionEntity, $notificationEntity, $context);
```

Comment:
Can we update the `handleSuccessfulNotification` and `handleFailedNotification` functions to return the OrderTransactionEntity, so that the `handleWebhook` returns the updated OrderTransactionEntity, instead of the same one passed to this function?

Reply:
Resolved since Shopware 6 does not return the updated orderTransactionEntity